File: backend/go-chi-hive/internal/cells/comment/domain.go
package comment

import (
	"time"
)

// Comment represents a comment on an article.
type Comment struct {
	ID        int64
	Body      string
	CreatedAt time.Time
	UpdatedAt time.Time
	Author    Author
}

// Author represents the comment's author profile view.
type Author struct {
	Username  string
	Bio       string
	Image     *string
	Following bool
}

File: backend/python-fastapi-vertical/app/features/comment/domain.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class CommentAuthor:
    username: str
    bio: str
    image: Optional[str]
    following: bool = False


@dataclass
class Comment:
    id: int
    created_at: datetime
    updated_at: datetime
    body: str
    author: CommentAuthor